tests/www/approvals_views/test_prolongation.py

```diff
@@ -133,6 +135,38 @@ def test_prolong_approval_view_bad_reason(self):
         )
         self.assertNotContains(response, self.MAX_DURATION_TEXT)
 
+    @freeze_time("2023-08-18")
+    def test_htmx_on_reason(self):
+        self.client.force_login(self.siae_user)
+        response = self.client.get(
+            reverse("approvals:declare_prolongation", kwargs={"approval_id": self.approval.pk}),
+        )
+        end_date = self.approval.end_at + relativedelta(days=30)
+        assert response.status_code == 200
+        htmx_soup = parse_response_to_soup(response, selector="#main")
+        data = {
+            "reason": ProlongationReason.RQTH,
+            # Workaround the validation of the initial page by providing enough data.
+            "end_at": end_date,
+            "email": self.prescriber.email,
+        }
+        response = self.client.post(
+            reverse("approvals:prolongation_form_for_reason", kwargs={"approval_id": self.approval.pk}),
+            data,
+        )
+        update_page_with_htmx(
+            htmx_soup,
+            "#id_reason_2",  # RQTH
+            response,
+        )
+        response = self.client.post(
+            reverse("approvals:declare_prolongation", kwargs={"approval_id": self.approval.pk}),
+            data,
+        )
+        assert response.status_code == 200
+        initial_soup = parse_response_to_soup(response, selector="#main")
+        assertSoupEqual(htmx_soup, initial_soup)
```

Comment:
Le nommage m'à fait des nœuds au cerveau :/, quand tu lis le test ça donne l'impression que les variables et leur contenu sont inversées sauf à la dernière ligne ^^".
```suggestion
        assertSoupEqual(via_htmx_soup, via_django_soup)
```
Même si je pense que ça ne résoudra pas le problème posé par `htmx_soup = parse_response_to_soup(...)` qui ne vient pas de HTMX mais qui est nécessaire pour être ensuite modifié _in place_ par `update_page_with_htmx()` :thinking:.

Reply:
J’ai renommé en `page` pour la page chargée à l’origine, modifiée par HTMX, et `fresh_page` pour la page obtenue après rechargement.